template.js

```diff
@@ -76,10 +83,46 @@ if (eventName === 'unknown') {
   return data.gtmOnFailure();
 }
 
+// Return the existing 'BillyPix' global method if available
+let BillyPix = copyFromWindow(billyFunctionName);
+
+function loadBackupBilly (BillyPixBackup) {
+  // Initialize BillyPix so the Tracking ID is set on the web page
+  BillyPixBackup('init', billyPixId);
+  
+  // Debug to see if correct
+  if (data.isDebug){
+    log('Backup: Successfully initialized the ' + billyFunctionName + ' for ID: ' + billyPixId);
+  }
+}
+
+function loadLibraryIfNotAvailable() {
+  // Function to ensure BillyPix is defined and properly queues commands
+  const BillyPixBackup = createArgumentsQueue(billyFunctionName, billyFunctionName + '.queue');
+  
+  // Setup BillyPix with the current timestamp
+  setInWindow(billyFunctionName + '.t', getTimestamp(), false);
+
+  // Generate the script URL with cache busting
+  const secondsBuste = 30*1000;
+  const epochRounded = secondsBuste * Math.ceil(getTimestamp() / secondsBuste);
+  const scriptUrl = cdnEndpoint + '?t=' + epochRounded + '&v=0.2.0';
+  
+  // Inject the BillyPix script
+  injectScript(scriptUrl, loadBackupBilly(BillyPixBackup), data.gtmOnFailure, scriptUrl);
+  
+  return BillyPixBackup;
+}
+
+
 // Sanity check: BillyPix needs to be available
 if (getType(BillyPix) === 'undefined') {
-  log(billyFunctionName + ' not available in window, make sure to first run the "Billy Grace - Web Configuration" tag');
-  return data.gtmOnFailure();
+    
+    // Make it clear they the user messed up
+    log(billyFunctionName + ' not available in window, make sure to first run the "Billy Grace - Web Configuration" tag. For now we are still loading it, but this can lead to performance issued');
```

Comment:
Nice!
Does this msg goes to GTM debug console only?
There is a way to let BG or client knows without GTML Debug mode?

Reply:
Yeah so this log will show up in the browser. Only unfortunately I could not find a way to make it a warning log with google's current APIs for web templates